contracts/OGNRewardsSource.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+import {Governable} from "./Governable.sol";
+import {Initializable} from "./upgrades/Initializable.sol";
+import "OpenZeppelin/openzeppelin-contracts@4.6.0/contracts/token/ERC20/IERC20.sol";
+
+contract OGNRewardsSource is Governable, Initializable {
+    error UnauthorizedCaller();
+    error RewardsTargetNotSet();
+    error InvalidRewardRate();
+
+    event StrategistUpdated(address _address);
+    event RewardsTargetChange(address target, address previousTarget);
+    event RewardsPerSecondChanged(uint256 newRPS, uint256 oldRPS);
+    event RewardCollected(uint256 amountCollected);
+
+    address public immutable ogn;
```

Comment:
Let's change `ogn` to `asset` since it will be more generic.

Reply:
Thought of that. But since the file name has "OGN" in it, I just went with `ogn` for this one as well. Will change it.

Also, IMO `rewardToken` might be a bit more apt than `asset`

Reply:
For the filename, what do you think of `RewardsSourceV2`?

Reply:
I'm good with `rewardToken`.

Maybe `fixedRewardsSource`? It's more of a different kind of thing, than an improved version.